### PR TITLE
feat: add GitHub Action to run test

### PR DESCRIPTION
**Pull request by trivikr**
Refs: https://github.com/facebook/jscodeshift/pull/491#issuecomment-1078550486

### CODE REVIEW

**Comment by trivikr** [comment]
@Daniel15 Can you enable GitHub Actions on jscodeshift repo so that this CI can run?

**Comment by Daniel15** [comment]
From reading our internal docs, it looks like GitHub Actions *should* be enabled by default, and I just need to merge these changes. Are the actions running on your fork (so we can see that they're working)?

**Comment by trivikr** [comment]
> Are the actions running on your fork (so we can see that they're working)?

Yes. I verified in https://github.com/trivikr/jscodeshift/pull/1 that GitHub Actions are running `yarn test`
* Node.js 12.x https://github.com/trivikr/jscodeshift/runs/5686361576
* Node.js 14.x https://github.com/trivikr/jscodeshift/runs/5686361625
* Node.js 16.x https://github.com/trivikr/jscodeshift/runs/5686361684

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        node-version: [12.x, 14.x, 16.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: yarn
```
Comment (Daniel15) [inline .github/workflows/test.yml:25]: Use `yarn --frozen-lockfile` please.

**Comment by Daniel15** [comment]
Awesome. Thanks! One small tweak inline and then I'll merge it.

File: .github/workflows/test.yml (hunk 1 above)
Comment (trivikr) [inline .github/workflows/test.yml:25]:
```suggestion
        run: yarn --frozen-lockfile
```

Comment (trivikr) [reply .github/workflows/test.yml]: Added in [`a980e68` (#493)](https://github.com/facebook/jscodeshift/pull/493/commits/a980e68108633290d2d95e320c19586c2c497e48)

Tested on fork:
* 12.x https://github.com/trivikr/jscodeshift/runs/5699022264?check_suite_focus=true
* 14.x https://github.com/trivikr/jscodeshift/runs/5699022321?check_suite_focus=true
* 16.x https://github.com/trivikr/jscodeshift/runs/5699022359?check_suite_focus=true

**Review by Daniel15 — APPROVED** [review]

**Pull request merged**